meson.build

```diff
@@ -26,6 +26,7 @@ add_project_arguments('-D_GNU_SOURCE', language: 'c')
 add_project_arguments('-Wmissing-prototypes', language: 'c')
 add_project_arguments('-Wno-format-truncation', language: 'c')
 add_project_arguments('-Wstrict-aliasing=2', language: 'c')
+add_project_arguments('-fms-extensions', language: 'c')
```

Comment:
Sorry, I thought it was just a troll so you would distract me from applying patches on DPDK for v24.11-rc1.

If it is not a troll, you'll have to convince me with an example of sane code that requires it.

Reply:
This was to have a struct inheritance mechanism.
```
struct foo {
    int a;
};
struct bar {
    struct foo;
    int b;
};
struct bar xzy = { .a = 1, .b = 2, };
```
Whereas you usually have to specify this:

```
struct bar {
    struct foo f;
    int b;
};

struct bar xzy = { .f.a = 1, .b = 2, };
```

Not a deal breaker, but as we wanted to have all priv data to have common fields, it was also to avoid duplication in the fields.
For instance, we have a `struct iface * input_iface` defined (and used ) in multiple locations (ipv4, ipv6) : this info should be placed by the rx node and reused across the graph.
As we don't have central location to define all private data, this gives a hint that the member is already present in the generic part of the private data.

I'll remove the compilation option, and define the generic priv data type as below:

```
#define GR_MBUF_PRIV_DATA_TYPE(type_name, fields)
        struct type_name {
                struct {
                        uint32_t flags;
                        struct gr_packet_traces traces;
                };
                struct fields;
        };
GR_MBUF_PRIV_DATA_TYPE(gr_mbuf, {});
```

Reply:
*cough* a macro would achieve the same *cough*
Let's go with the simple duplication approach for now.

Reply:
yup, modified in v2.